doc/api/vm.md

```diff
@@ -10,36 +10,75 @@ The `vm` module enables compiling and running code within V8 Virtual
 
-JavaScript code can be compiled and run immediately or
-compiled, saved, and run later.
+JavaScript code can be compiled and run immediately or compiled, saved, and run
+later.
 
-A common use case is to run the code in a different V8 Context. This means
-invoked code has a different global object than the invoking code.
-
-One can provide the context by [_contextifying_][contextified] an
-object. The invoked code treats any property in the context like a
-global variable. Any changes to global variables caused by the invoked
-code are reflected in the context object.
+A common use case is to run code within a different environment. Each context
+uses a different V8 Context, meaning that it has a different global object than
+the rest of the code. One can provide a context by using the [`vm.Context`][]
+constructor.
 
 ```js
 const vm = require('vm');
 
 const x = 1;
 
-const context = { x: 2 };
-vm.createContext(context); // Contextify the object.
+const context = new vm.Context();
+context.global.x = 2;
 
 const code = 'x += 40; var y = 17;';
-// `x` and `y` are global variables in the context.
-// Initially, x has the value 2 because that is the value of context.x.
-vm.runInContext(code, context);
+// `x` and `y` are global variables in the context environment.
+// Initially, x has the value 2 because that is the value of context.global.x.
+const script = new vm.Script(code);
+script.runInContext(context);
 
-console.log(context.x); // 42
-console.log(context.y); // 17
+console.log(context.global.x); // 42
+console.log(context.global.y); // 17
 
 console.log(x); // 1; y is not defined.
 ```
 
+## Class: `vm.Context`
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+### Constructor: `new vm.Context([options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `options` {Object}
+  * `name` {string} Human-readable name of the newly created context.
+    **Default:** `'VM Context i'`, where `i` is an ascending numerical index of
+    the created context.
+  * `origin` {string} [Origin][origin] corresponding to the newly created
+    context for display purposes. The origin should be formatted like a URL,
+    but with only the scheme, host, and port (if necessary), like the value of
+    the [`url.origin`][] property of a [`URL`][] object. Most notably, this
+    string should omit the trailing slash, as that denotes a path.
+    **Default:** `''`.
+  * `codeGeneration` {Object}
+    * `strings` {boolean} If set to false any calls to `eval` or function
+      constructors (`Function`, `GeneratorFunction`, etc) will throw an
+      `EvalError`. **Default:** `true`.
+    * `wasm` {boolean} If set to false any attempt to compile a WebAssembly
+      module will throw a `WebAssembly.CompileError`. **Default:** `true`.
+
+Create a new VM Context. This context will have a unique global object and
+variable environment.
+
+### `context.global`
+<!-- YAML
+added: REPLACEME
+-->
+
+* {Object}
+
+The global object of this context.
```

Comment:
Is this the actual global object or merely the global `this` value / global proxy?

If it’s the former, then it would be better to document it clearly (perhaps with appropriate links to external resources), as I doubt most Node.js users have detailed knowledge on the difference. (I.e., most people will probably say that `window` is the global object.)

If it’s the latter (and I assume this to be the case), then we should call it `globalThis` in accordance with the global property. I think we could even omit this API as one could easily evaluate `this` inside the context to get the value.

Reply:
It's the global proxy I think?  I can change the name I suppose.

> I think we could even omit this API as one could easily evaluate this inside the context to get the value.

This is something that I explicitly want to not be needed.

Reply:
It is the global proxy: <https://github.com/nodejs/node/blob/45c5232d9baf133f692fc300437eebac680f12a0/src/node_contextify.cc#L296>

Arguably, the name should be `globalProxy`.

---

Also, for **JSDOM**, it’d be helpful to have a reference to the underlying `globalObject` as well.